libs/crypto/include/crypto/openssl_hasher.hpp

```diff
@@ -53,15 +53,17 @@ class OpenSslHasher
   OpenSslHasher &operator=(OpenSslHasher const &) = delete;
   OpenSslHasher &operator=(OpenSslHasher &&) = delete;
 
-  bool        reset();
-  bool        update(uint8_t const *data_to_hash, std::size_t size);
-  bool        final(uint8_t *data_to_hash);
-  std::size_t hash_size() const;
+  bool        Reset();
+  bool        Update(uint8_t const *data_to_hash, std::size_t size);
+  bool        Final(uint8_t *data_to_hash);
+  std::size_t HashSize() const;
 
 private:
-  internal::OpenSslHasherImpl *impl_;
+  std::shared_ptr<internal::OpenSslHasherImpl> impl_;
 };
 
+inline OpenSslHasher::~OpenSslHasher() = default;
```

Comment:
Again think this can be moved into the declaration.

Reply:
:+1: